common/utils/src/test/scala/org/apache/spark/util/StructuredLoggingSuite.scala

```diff
@@ -438,6 +438,44 @@ class StructuredLoggingSuite extends LoggingSuiteBase {
     assert((log"\r" + log"\n" + log"\t" + log"\b").message == "\r\n\t\b")
     assert((log"\r${MDC(LogKeys.EXECUTOR_ID, 1)}\n".message == "\r1\n"))
   }
+
+  test("disabled structured logging won't log context") {
+    Logging.disableStructuredLogging()
+    val expectedPatternWithoutContext = compactAndToRegexPattern(
+      s"""
+        {
+          "ts": "<timestamp>",
+          "level": "INFO",
+          "msg": "Lost executor 1.",
+          "logger": "$className"
+        }""")
+
+    Seq(
+      () => logInfo(log"Lost executor ${MDC(LogKeys.EXECUTOR_ID, "1")}."),
+      () => logInfo( // blocked when explicitly constructing the MessageWithContext
+        MessageWithContext(
+          "Lost executor 1.",
+          new java.util.HashMap[String, String] { put(LogKeys.EXECUTOR_ID.name, "1") }
```

Comment:
```
  private[spark] val STRUCTURED_LOGGING_ENABLED =
    ConfigBuilder("spark.log.structuredLogging.enabled")
      .doc("When true, Spark logs are output as structured JSON lines with added Spark " +
        "Mapped Diagnostic Context (MDC), facilitating easier integration with log aggregation " +
        "and analysis tools. When false, logs are plain text without MDC. This configuration " +
        "does not apply to interactive environments such as spark-shell, spark-sql, and " +
        "PySpark shell.")
      .version("4.0.0")
      .booleanConf
      .createWithDefault(true)
```
Disabling the structured logging will disable the MDCs. This is on purpose to avoid regression on setting these contexts.

Reply:
The MDC context is set in `withLogContext`, checking the config value there should meet the semantics of this config. But I can add changes in lines 154 - 156 back to even avoid setting those in the data structure!